pkg/metrics/statsd/cosmosdb/metrics.go

```diff
@@ -0,0 +1,75 @@
+package cosmodb
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/metrics"
+)
+
+var _ http.RoundTripper = (*tracerRoundTripper)(nil)
+
+type tracerRoundTripper struct {
+	log *logrus.Entry
+	m   metrics.Interface
+	tr  http.RoundTripper
+}
+
+func New(log *logrus.Entry, tr *http.Transport, m metrics.Interface) *tracerRoundTripper {
+	return &tracerRoundTripper{
+		log: log,
+		m:   m,
+		tr:  tr,
+	}
+}
+
+func (t *tracerRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	start := time.Now()
+	resp, err := t.tr.RoundTrip(r)
+	if err != nil {
+		return nil, err
+	}
+
+	var ru float64
+	// Sometimes we get request-charge="" because pkranges API is free
+	// We log this on debug mode only and ignore
+	if resp.Header.Get("x-ms-request-charge") != "" &&
+		resp.Header.Get("x-ms-request-charge") != `""` {
+		f, err := strconv.ParseFloat(strings.Trim(resp.Header.Get("x-ms-request-charge"), "\""), 64)
+		if err != nil {
+			// we don't want to kill all DB call if this changes
+			t.log.Errorf("error: %v", err)
+			err = nil
+		} else {
+			ru = f
+		}
+	}
+
+	t.m.EmitGauge("client.cosmosdb.count", 1, map[string]string{
+		"code": strconv.Itoa(resp.StatusCode),
+		"verb": r.Method,
+		"path": r.URL.Path,
```

Comment:
the cardinality of r.URL.Path is too high in the case of get, replace and delete of documents: the path will end "/docs/<unique-id>".  I don't mind db/coll/trigger ids appearing as they are effectively low cardinality, but we can't do it for docs ids.  Need the following:

```
	parts := strings.Split(r.URL.Path, "/")
	if len(parts) >= 2 && parts[len(parts)-2] == "docs" {
		parts[len(parts)-1] = "{id}"
	}
	path := strings.Join(parts, "/")
```